// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Message representing the waveform.

syntax = "proto3";

package eidos.audition;

// Next available ID: 5
message Waveform {
  // Number of channels.
  int32 num_channels = 1;

  // Sample rate (in Hz).
  int32 sample_rate = 2;

  // Number of bits per sample.
  int32 bits_per_sample = 3;

  // Samples.
  repeated float samples = 4;
}

// Local Variables:
// mode: c++
// End:
